client/adjudicate.go

```diff
@@ -417,23 +417,34 @@ func (c *Channel) ensureRegistered(ctx context.Context) error {
 		return nil
 	}
 
-	registeredEvents := make(chan *channel.RegisteredEvent)
-
+	events := make(chan channel.AdjudicatorEvent)
 	// Start event subscription.
 	sub, err := c.adjudicator.Subscribe(ctx, c.Params().ID())
 	if err != nil {
 		return errors.WithMessage(err, "subscribing to adjudicator events")
 	}
-	go waitForRegisteredEvent(sub, registeredEvents)
+	defer func() {
+		if err := sub.Close(); err != nil {
```

Comment:
The only error that a `Closer` can return from its `Close` function is an `AlreadyClosedError`.  
Maybe we should either: Rethink the error logic or/and add logging to the `Close` method to not bloat our `defer sub.Close()` calls.

Reply:
I thought the same while looking at my code. This is kind of ugly currently. However, it's a separate issue, hence, let's address it in a separate PR.

Reply:
referenced in #211